src/main/java/org/venice/beachfront/bfapi/controllers/UserExceptionHandler.java

```diff
@@ -33,7 +33,7 @@
 import model.logger.Severity;
 import util.PiazzaLogger;
 
-@ControllerAdvice
+@ControllerAdvice(basePackages = "org.venice.beachfront.bfapi.controllers")
```

Comment:
What is the purpose of this line? Is there an instance where we don't want to globally apply this Controller Advice? OAuth related?

Reply:
When authenticating with OAuth, one of the internal OAuth classes throws an Exception that it's own OAuth ExceptionHandler needs to catch and re-route for authentication. However, this `UserExceptionHandler` was catching the Exception instead and breaking the authentication flow. By including the specific controller classes via the `basePackages` field, the OAuth exception doesn't get caught by the `UserExceptionHandler` and authentication can proceed.